lib/actions/local/common.js

```diff
@@ -0,0 +1,99 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as dockerUtils from '../../utils/docker';
+import { exitWithExpectedError } from '../../utils/patterns';
+import { getChalk } from '../../utils/lazy';
+
+export const dockerPort = 2375;
+export const dockerTimeout = 2000;
+
+export const filterOutSupervisorContainer = function(container) {
+	for (let name of container.Names) {
+		if (
+			name.includes('resin_supervisor') ||
+			name.includes('balena_supervisor')
+		) {
+			return false;
+		}
+	}
+	return true;
+};
+
+export const selectContainerFromDevice = Promise.method(function(
+	deviceIp,
+	filterSupervisor,
+) {
+	if (filterSupervisor == null) {
+		filterSupervisor = false;
+	}
+	const form = require('resin-cli-form');
+	const docker = dockerUtils.createClient({
+		host: deviceIp,
+		port: dockerPort,
+		timeout: dockerTimeout,
+	});
+
+	// List all containers, including those not running
+	return docker.listContainers({ all: true }).then(function(containers) {
+		containers = containers.filter(function(container) {
+			if (!filterSupervisor) {
+				return true;
+			}
+			return filterOutSupervisorContainer(container);
+		});
```

Comment:
I switched this from a Bluebird.filter to an Array.filter as the native dockerode promises are not typed as Bluebird promises (they are in this case, but just having a generic promise type) so typescript complained, and we didn't actually need it to be a Bluebird.filter so it should be more performant/clearer with the new version